File: flutter/lib/features/weather_search/presentation/cubit/weather_error_message.dart
import '../../domain/entities/weather_exception.dart';

/// 將領域錯誤類型轉為要顯示給使用者的訊息（presentation 文案）。與 iOS 端一致。
class WeatherErrorMessage {
  const WeatherErrorMessage._();

  static String text(WeatherErrorType type) {
    switch (type) {
      case WeatherErrorType.cityNotFound:
        return '查無此城市，請確認名稱後再試';
      case WeatherErrorType.invalidData:
        return '資料格式不正確，請稍後再試';
      case WeatherErrorType.requestFailed:
        return '連線失敗，請檢查網路或稍後再試';
    }
  }
}

File: ios-native/weatherPrediction/Features/WeatherSearch/ViewModel/WeatherErrorMessage.swift
import DomainModule
import Foundation

/// 將領域錯誤轉為要顯示給使用者的訊息（presentation 文案）。
enum WeatherErrorMessage {
    static func text(for error: WeatherError) -> String {
        switch error {
        case .cityNotFound:
            return "查無此城市，請確認名稱後再試"
        case .invalidData:
            return "資料格式不正確，請稍後再試"
        case .requestFailed:
            return "連線失敗，請檢查網路或稍後再試"
        case .unauthorized:
            return "授權失敗，請確認 API 授權碼"
        }
    }
}
